### ISSUE TITLE
Add theory section on extrapolation methods 

### ISSUE THREAD

**Issue by purva-thakre** [unitaryfund/mitiq#491]
It could be useful if there's a section describing what all the specified extrapolation methods do before[ this section](https://mitiq.readthedocs.io/en/v0.1.0/guide/guide-zne.html#classical-fitting-and-extrapolation-factory-objects) in documentation.

**Comment by willzeng** [comment]
Agreed this would be a good "living" update to the descriptions already in the arXiv papers

**Comment by purva-thakre** [comment]
Don't think I will have the chance to do this soon. If anyone wants to assign it to themselves, feel free to do so.
